### DISCUSSION TITLE
Installation directory setting

### DISCUSSION THREAD

**Discussion by connorourke** [fppimenta/rheoTool#54] [Q&A] [answered]
In the User guide its says, with reference to the installation folder:

 `A good location for rheoTool is, for example, directory $WM PROJECT USER DIR, which is defined by default when OpenFOAM⃝R or foam-extend is installed.` 

It isn't clear how to define the installation folder. At the moment it gets put in `$FOAM_USER_APPBIN`. 

How do I tell Allwmake where to put rheoTool?

Thanks

**Comment by fppimenta** [comment]
One thing is the directory to where you clone/download the repo, which will contain the code source, user-guide and tutorials and from which you run the installer. Your transcription of the user-guide refers to that. Another thing is the directory to which the binaries (apps and libs) are written, which are $FOAM_USER_APPBIN and $FOAM_USER_LIBBIN. This is a logic choice, since they both are sourced by default by openfoam. Can you change them? Yes, in all Make/files included in the repo, but you will then have to tell the compiler and loader where to find them, unless they are already included by default.

**Reply by connorourke** [reply]
Thanks for the answer - so to be clear: to define the installation directory all I need to do is to export the required directories to $FOAM_USER_APPBIN and $FOAM_USER_LIBBIN before compiling, and then make sure they are included in the $PATH and $LD_LIBRARY_PATH?

**Comment by fppimenta** [comment, marked as answer]
Yes, it is more or less like that. More info here: https://cfd.direct/openfoam/user-guide/v6-compiling-applications/.

Note: you just need to proceed as with any other c++ program, but instead of defining things in a makefile or in the terminal, you define them in Make/options and Make/files of each lib/app. Thus, you just need to understand how things are defined there.

**Reply by connorourke** [reply]
Great - thanks.
